src/app/pages/rpc-sign-stacks-message/use-rpc-sign-stacks-message.ts

```diff
@@ -0,0 +1,157 @@
+import { useMemo } from 'react';
+import { useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { RpcErrorCode } from '@btckit/types';
+import { StacksNetwork } from '@stacks/network';
+import { deserializeCV } from '@stacks/transactions';
+
+import { logger } from '@shared/logger';
+import { makeRpcErrorResponse, makeRpcSuccessResponse } from '@shared/rpc/rpc-methods';
+import {
+  UnsignedMessage,
+  isSignableMessageType,
+  isStructuredMessageType,
+  isUtf8MessageType,
+} from '@shared/signature/signature-types';
+import { closeWindow } from '@shared/utils';
+
+import { useAnalytics } from '@app/common/hooks/analytics/use-analytics';
+import { useDefaultRequestParams } from '@app/common/hooks/use-default-request-search-params';
+import { useRejectIfLedgerWallet } from '@app/common/rpc-helpers';
+import { useWalletType } from '@app/common/use-wallet-type';
+import {
+  StructuredPayload,
+  Utf8Payload,
+} from '@app/features/stacks-message-signer/stacks-message-signing';
+import {
+  improveUxWithShortDelayAsStacksSigningIsSoFast,
+  useMessageSignerStacksSoftwareWallet,
+} from '@app/features/stacks-message-signer/stacks-message-signing.utils';
+
+function getNetwork(networkName: string | null) {
+  if (
+    networkName === 'mainnet' ||
+    networkName === 'testnet' ||
+    networkName === 'devnet' ||
+    networkName === 'mocknet'
+  ) {
+    return StacksNetwork.fromName(networkName);
+  }
+  return;
+}
+
+export function useRpcPayload() {
```

Comment:
Let's rename this to something specific to the feature `useRpcStacksMessagePayload`?